sdk/sdk-core/aar/src/main/java/mil/emp3/core/mapgridlines/AbstractMapGridLine.java

```diff
@@ -0,0 +1,376 @@
+            try {
+                this.join();
+            } catch (InterruptedException e) {
+            }
+        }
+    }
+
+    protected AbstractMapGridLine(IMapInstance mapInstance) {
+        this.strokeStyleMap = new HashMap<>();
+        this.labelStyleMap = new HashMap<>();
+        this.lastUpdated = new Date();
+        this.currentCamera = new Camera();
+        this.previousCamera = new Camera();
+        this.boundingBox = new EmpBoundingBox();
+        this.mapInstance = mapInstance;
+        this.generationThread = new GridLineGenerationThread();
+        this.generationThread.setPriority(this.generationThread.getPriority() + 1);
+        this.generationThread.start();
+
+        setStyles();
+    }
+
+    // This method loads the styles.
+    private void setStyles() {
+        EmpGeoColor color;
+        IGeoLabelStyle labelStyle;
+
+        // MGRS Grid Label
+        labelStyle = new GeoLabelStyle();
+        color = new EmpGeoColor(1.0, 200, 200, 200);
+        labelStyle.setColor(color);
+        labelStyle.setSize(8.0);
+        labelStyle.setJustification(IGeoLabelStyle.Justification.CENTER);
+        labelStyle.setFontFamily("Ariel");
+        labelStyle.setTypeface(IGeoLabelStyle.Typeface.BOLD);
+        addLabelStyle(MAIN_GRID_TYPE_LABEL, labelStyle);
+    }
+
+    @Override
+    public void shutdownGenerator() {
+        if (null != this.generationThread) {
+            this.generationThread.exitThread();
+            this.generationThread = null;
+        }
+    }
+
+    @Override
+    public Date getLastUpdated() {
+        return this.lastUpdated;
+    }
+
+    @Override
+    public List<IFeature> getGridFeatures() {
+        // We need to return the feature in the return list so the map can loop thru it with out us
+        // runing into a multi threading issue.
+        synchronized (this.featureList) {
+            this.returnList.clear();
+            if (!this.featureList.isEmpty()) {
+                this.returnList.addAll(this.featureList);
+            }
+        }
+        return this.returnList;
+    }
+
+    protected void clearFeatureList() {
+        boolean resetTime = (this.featureList.size() > 0);
+        synchronized (this.featureList) {
+            this.featureList.clear();
+        }
+
+        if (resetTime) {
+            this.lastUpdated.setTime(System.currentTimeMillis());
+        }
+    }
+
+    protected void addFeature(IFeature feature) {
+        if (null == feature) {
+            throw new InvalidParameterException("feature is null.");
+        }
+        synchronized (this.featureList) {
+            this.featureList.add(feature);
+        }
+        this.lastUpdated.setTime(System.currentTimeMillis());
+    }
+
+    @Override
+    public void mapViewChange(IGeoBounds mapBounds, ICamera camera, int viewWidth, int viewHeight) {
+        if ((null == mapBounds) || (null == camera)) {
+            clearFeatureList();
+            this.mapInstance.scheduleMapRedraw();
```

Comment:
I no longer use the camera but it need the bounds. If there are no bounds, the grid is removed. I.E. when the camera is looking up. I will add a comment.